### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by puneet-pdx**
Related to issue: #

### Description:

Update 3rd Party dependencies

### Summary of changes:

- update 3rd party dependencies
- update gradle plugin
- resolve compilation issues 
- resolve lint issues

### Pre-merge Checklist

- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/) Job for this PR has been run
  - [x] link: https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/228/
  - [ ] https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/237/
- Unit and/or integration tests have been added to exercise this PR's logic, and the tests are passing:
  - [ ] Yes
  - [ ] No

### CODE REVIEW

File: toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/GeoViewScope.kt [hunk 1]
```diff
@@ -723,30 +723,29 @@ private fun Modifier.drawCalloutContainer(
...
-                    color = backgroundColor,
-                    style = Fill
-                )
-                // Outline the path's shape with the Callout's stroke color
-                drawPath(
-                    path = path,
-                    color = strokeColor,
-                    style = Stroke(width = strokeBorderWidth)
-                )
-            }
```
Comment (puneet-pdx) [inline toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/GeoViewScope.kt:746]: Resolves Lint error 

Using Modifier.then with a Modifier factory function with an implicit receiver More... (⌘F1) 
Inspection info: Calling a Modifier factory function with an implicit receiver inside Modifier.then will result in the receiver (this) being added twice to the chain. For example, fun Modifier.myModifier() = this.then(otherModifier()) - the implementation of factory functions such as Modifier.otherModifier() will internally call this.then(...) to chain the provided modifier with their implementation. When you expand this.then(otherModifier()), it becomes: this.then(this.then(OtherModifierImplementation)) - so you can see that this is included twice in the chain, which results in modifiers such as padding being applied twice, for example. Instead, you should either remove the then() and directly chain the factory function on the receiver, this.otherModifier(), or add the empty Modifier as the receiver for the factory, such as this.then(Modifier.otherModifier())  Issue id: SuspiciousModifierThen Vendor: Jetpack Compose Identifier: androidx.compose.ui Feedback: https://issuetracker.google.com/issues/new?component=612128

File: toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/GeoViewScope.kt [hunk 2]
```diff
@@ -723,30 +723,29 @@ private fun Modifier.drawCalloutContainer(
     leaderWidth: Float,
     leaderHeight: Float,
     minSize: DpSize
```
Comment (puneet-pdx) [inline toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/GeoViewScope.kt:725]: Resolves Lint error 

Using Modifier.then with a Modifier factory function with an implicit receiver More... (⌘F1) 
Inspection info: Calling a Modifier factory function with an implicit receiver inside Modifier.then will result in the receiver (this) being added twice to the chain. For example, fun Modifier.myModifier() = this.then(otherModifier()) - the implementation of factory functions such as Modifier.otherModifier() will internally call this.then(...) to chain the provided modifier with their implementation. When you expand this.then(otherModifier()), it becomes: this.then(this.then(OtherModifierImplementation)) - so you can see that this is included twice in the chain, which results in modifiers such as padding being applied twice, for example. Instead, you should either remove the then() and directly chain the factory function on the receiver, this.otherModifier(), or add the empty Modifier as the receiver for the factory, such as this.then(Modifier.otherModifier())  Issue id: SuspiciousModifierThen Vendor: Jetpack Compose Identifier: androidx.compose.ui Feedback: https://issuetracker.google.com/issues/new?component=612128

**Review by TADraeseke — APPROVED** [review]
Tested on a few different scene/map micro apps. All looks fine!

**Review by eri9000 — CHANGES_REQUESTED** [review]
@puneet-pdx 
I am getting an error when building the project. It seems that some of the libraries are not compatible with `MircroapsLib->compilesdk = 34`. Changing it to 35 seems to fix it

**Review by eri9000 — APPROVED** [review]

**Pull request merged**